games-fps/eduke32/eduke32-20180508.6885.ebuild

```diff
@@ -0,0 +1,236 @@
+	psx? ( ^^ ( demo cdinstall ) )
+	sc-55? ( ^^ ( demo cdinstall )
+		!opl )
+	vpx? ( opengl )"
+
+S="${WORKDIR}/${PN}_${MY_DATE}-${MY_BUILD}"
+
+MY_DEPEND_RDEPEND="media-sound/timidity-freepats
+	sys-libs/zlib:=
+	flac? ( media-libs/flac )
+		gtk? ( x11-libs/gtk+:2 )
+	opengl? ( virtual/glu
+		virtual/opengl )
+	png? ( media-libs/libpng:0= )
+	sdl1? ( media-libs/libsdl[joystick,opengl?,sound,video,X]
+		media-libs/sdl-mixer[flac?,midi,timidity,vorbis?] )
+	!sdl1? ( media-libs/libsdl2[joystick,opengl?,sound,video,X]
+		media-libs/sdl2-mixer[flac?,midi,timidity,vorbis?] )
+	vpx? ( media-libs/libvpx:= )
+	vorbis? ( media-libs/libogg
+		media-libs/libvorbis )
+	xmp? ( media-libs/exempi:= )"
+
+RDEPEND="${MY_DEPEND_RDEPEND}
+	cdinstall? ( games-fps/duke3d-data )
+	demo? ( games-fps/duke3d-demodata )"
+
+DEPEND="${MY_DEPEND_RDEPEND}
+	app-arch/unzip
+	x86? ( dev-lang/nasm )"
+
+PATCHES=( "${FILESDIR}/fix-build-transpal.patch" "${FILESDIR}/log-to-tmpdir.patch" "${FILESDIR}/search-duke3d-path.patch" )
+
+src_unpack() {
+	# Extract only the eduke32 archive
+	unpack ${PN}_src_${MY_DATE}-${MY_BUILD}.tar.xz
+
+	# Unpack only the documentation
+	if use hrp; then
+		unzip -q "${DISTDIR}"/${MY_PN_HRP}-${MY_PV_HRP}.zip hrp_readme.txt hrp_todo.txt || die
+	fi
+	if use offensive; then
+		unzip -q "${DISTDIR}"/${MY_PN_XXX}-${MY_PV_XXX}.zip xxx_readme.txt || die
+	fi
+	if use opl; then
+		unzip -q "${DISTDIR}"/${MY_PN_OPL}-${MY_PV_OPL}.zip readme.txt || die
+	fi
+	if use sc-55; then
+		unzip -q "${DISTDIR}"/${MY_PN_SC55}-${MY_PV_SC55}.zip readme/music_readme.txt || die
+	fi
+}
+
+src_compile() {
+	local myemakeopts=(
+		ALLOCACHE_AS_MALLOC=0
+		AS=$(tc-getAS)
+		CC=$(tc-getCC)
+		CLANG=0
+		CPLUSPLUS=1
+		CUSTOMOPT=""
+		DEBUGANYWAY=0
+		F_JUMP_TABLES=""
+		FORCEDEBUG=0
+		HAVE_FLAC=$(usex flac 1 0)
+		HAVE_GTK2=$(usex gtk 1 0)
+		HAVE_VORBIS=$(usex vorbis 1 0)
+		HAVE_XMP=$(usex xmp 1 0)
+		LINKED_GTK=$(usex gtk 1 0)
+		LTO=1
+		LUNATIC=0
+		KRANDDEBUG=0
+		MEMMAP=0
+		MIXERTYPE=SDL
+		NETCODE=$(usex server 1 0)
+		NOASM=0
+		OPTLEVEL=0
+		OPTOPT=""
+		PACKAGE_REPOSITORY=1
+		POLYMER=$(usex opengl 1 0)
+		PRETTY_OUTPUT=0
+		PROFILER=0
+		RELEASE=1
+		RENDERTYPE=SDL
+		SDL_TARGET=$(usex sdl1 1 2)
+		SIMPLE_MENU=0
+		STRIP=""
+		TANDALONE=0
+		STARTUP_WINDOW=$(usex gtk 1 0)
+		USE_OPENGL=$(usex opengl 1 0)
+		USE_LIBVPX=$(usex vpx 1 0)
+		USE_LIBPNG=$(usex png 1 0)
+		USE_LUAJIT_2_1=0
+		WITHOUT_GTK=$(usex gtk 0 1)
+	)
+
+	emake "${myemakeopts[@]}"
+
+	if use tools; then
+		emake utils "${myemakeopts[@]}"
+	fi
+}
+
+src_install() {
+	local binary
+	local binaries=(
+		"eduke32"
+		"mapster32"
+		"${FILESDIR}/eduke32-bin"
+	)
+	for binary in "${binaries[@]}"; do
+		dobin ${binary}
+	done
+
+	local tool
+	local tools=(
+		"arttool"
+		"bsuite"
+		"cacheinfo"
+		"generateicon"
+		"givedepth"
+		"ivfrate"
+		"kextract"
+		"kgroup"
+		"kmd2tool"
+		"makesdlkeytrans"
+		"map2stl"
+		"md2tool"
+		"mkpalette"
+		"transpal"
+		"unpackssi"
+		"wad2art"
+		"wad2map"
+	)
```

Comment:
Do we really need quotes around all these?

Reply:
No, I've removed them now. But IMHO the don't harm.